tests/conftest.py

```diff
@@ -0,0 +1,27 @@
+# conftest.py
+def pytest_collection_modifyitems(session, config, items):
+    """Modifies test items to ensure test functions run in a given order"""
```

Comment:
The motivation is to test the most basic things first before moving on to more complex tests. That way if there's an issue with something basic, it'll take fewer completed tests before uncovering the issue. This isn't a big deal for GitHub's automated testing, but I semi-regularly run the tests offline while developing, and this will let me interrupt the test run quicker. It also just seems like best practice to test smaller components before testing larger integrated sections of code.

Reply:
This also isn't a user-facing function btw; it modifies the order of tests when you do `pytest tests`